lib/ramble/ramble/application.py

```diff
@@ -530,22 +530,37 @@ def build_used_variables(self, workspace):
         self.build_modifier_instances()
         self.add_expand_vars(workspace)
 
+        backup_variables = self.variables.copy()
+
+        ########################
+        # Define extra variables
+        ########################
+
+        # Add modifier mode variables defaults as a placeholder:
+        for mod_inst in self._modifier_instances:
+            for var in mod_inst.mode_variables().values():
+                if var.name not in self.variables:
+                    self.variables[var.name] = var.default
+
+        if self.package_manager is not None:
+            # Add package manager variable defaults as placeholder
+            for var in self.package_manager.package_manager_variables.values():
+                self.expander._variables[var.name] = var.default
```

Comment:
Curious why pkgman vars are added to expander, while modifiers are added to `self.variables`?

Reply:
Silly typo. They are both references to the same thing. So, which one they are added to is not super important. I'll fix this though.